dungeon/src/task/quizquestion/UIAnswerCallback.java

```diff
@@ -0,0 +1,117 @@
+
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.CheckBox;
+import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
+import com.badlogic.gdx.scenes.scene2d.ui.TextArea;
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup;
+import com.badlogic.gdx.utils.SnapshotArray;
+
+import core.Entity;
+import core.Game;
+import core.hud.TextDialog;
+import core.hud.UITools;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+/**
+ * Contains functions to build an easy-to-use Consumer as a callback for the {@link
+ * contrib.components.InteractionComponent}.
+ *
+ * <p>The Consumer will show the given {@link Quiz} on the HUD and will call the (as parameter)
+ * given Callback with the given answers as a {@link Quiz.Content}-Set.
+ *
+ * <p>Use {@link #askOnInteraction(Quiz, Consumer)} to get the consumer that you can use as an
+ * interaction callback.
+ */
+public final class UIAnswerCallback {
+
+    /**
+     * Get a Consumer that can be used as a callback for the {@link
+     * contrib.components.InteractionComponent} and will show the given {@link Quiz} on the UI.
+     *
+     * @param quiz The {@link Quiz} to show on the HUD on interaction.
+     * @param dslCallback Callback function that will be called after the player confirms their
+     *     answers.
+     * @return Consumer to use as a callback for the interaction component.
+     */
+    public static Consumer<Entity> askOnInteraction(
+            Quiz quiz, Consumer<Set<Quiz.Content>> dslCallback) {
+        return questGiver ->
+                QuizUI.showQuizDialog(
+                        quiz, (Entity hudEntity) -> uiCallback(quiz, hudEntity, dslCallback));
+    }
+
+    /**
+     * Callback after the answers where confirmed.
+     *
+     * @see UITools
+     */
+    private static BiFunction<TextDialog, String, Boolean> uiCallback(
+            Quiz quest, Entity hudEntity, Consumer<Set<Quiz.Content>> dslCallback) {
+        return (textDialog, id) -> {
+            if (Objects.equals(id, UITools.DEFAULT_DIALOG_CONFIRM)) {
+                dslCallback.accept(getAnswer(quest, answerSection(textDialog)));
+                Game.removeEntity(hudEntity);
+                return true;
+            }
+            return false;
+        };
+    }
+
+    private static VerticalGroup answerSection(TextDialog textDialog) {
+        SnapshotArray<Actor> children =
+                ((VerticalGroup) textDialog.getContentTable().getChildren().get(0)).getChildren();
+        // find the answer section
+        return (VerticalGroup)
+                children.select((actor) -> Objects.equals(actor.getName(), ANSWERS_GROUP_NAME))
+                        .iterator()
+                        .next();
+    }
+
+    private static Set<Quiz.Content> getAnswer(Quiz quiz, VerticalGroup answerSection) {
+        if (quiz.type() == Quiz.Type.FREETEXT) {
```

Comment:
@malt-r hier bin ich mir irgendwie nicht so sicher.
Für SC und MC Antworten erstellst du über die DSL `Quiz.Content` Instanzen, diese kann ich dir raussuchen und zurückgeben. 
Für Freitext Aufgaben erstellst du in der DSL keine `Quiz.Content` Instanz, da ist ja schließlich auch nichts.
Meine erste Idee war es jetzt, die abgegebene Antwort einfach eigenständig zu einer `Quiz.Content` Instanzen zu machen und dir diese dann im Callback zurück zu geben. 

Jetzt kennst du das Ding aber in der DSL ja nicht, oder? Ist das schlimm? Oder brauchst du das gar nicht? Oder ziehst du dir das eh aus dem `Task`?

Reply:
im meeting besprochen